modules/pricer/src/main/java/com/opengamma/strata/pricer/impl/option/SABRInterestRateParameters.java

```diff
@@ -0,0 +1,576 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.impl.option;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanBuilder;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.PutCall;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.tuple.DoublesPair;
+import com.opengamma.strata.market.surface.NodalSurface;
+import com.opengamma.strata.math.impl.function.Function1D;
+import com.opengamma.strata.pricer.impl.volatility.VolatilityModel;
+import com.opengamma.strata.pricer.impl.volatility.smile.function.SABRFormulaData;
+import com.opengamma.strata.pricer.impl.volatility.smile.function.VolatilityFunctionProvider;
+
+/**
+ * The volatility surface description under SABR model.  
+ * <p>
+ * This is used in interest rate modeling. 
+ * Each SABR parameter is {@link NodalSurface} spanned by expiration and tenor. 
+ */
+@BeanDefinition(builderScope = "private")
+public final class SABRInterestRateParameters
+    implements VolatilityModel<double[]>, ImmutableBean {
+
+  /**
+   * The alpha (volatility level) surface. 
+   * <p>
+   * The first dimension is the expiration and the second the tenor.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final NodalSurface alphaSurface;
+  /**
+   * The beta (elasticity) surface. 
+   * <p>
+   * The first dimension is the expiration and the second the tenor.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final NodalSurface betaSurface;
+  /**
+   * The rho (correlation) surface. 
+   * <p>
+   * The first dimension is the expiration and the second the tenor.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final NodalSurface rhoSurface;
+  /**
+   * The nu (volatility of volatility) surface. 
+   * <p>
+   * The first dimension is the expiration and the second the tenor.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final NodalSurface nuSurface;
+  /**
```

Comment:
We should probably incorporate the "shift" in this version. I don't think there is value to have one model with SABR and one with "shifted SABR". It is enough to have a constructor that create a shift of a constant 0 for the simpler case.